rest/admin_api.go

```diff
@@ -134,12 +134,15 @@ func (h *handler) handlePutDbConfig() error {
 	if err != nil {
 		return err
 	}
+	config.Server = &h.server.config.Bootstrap.Server
 	if err := config.setup(dbName); err != nil {
 		return err
 	}
 	h.server.lock.Lock()
 	defer h.server.lock.Unlock()
-	h.server.config.Databases[dbName] = config
+	dbc := DatabaseConfig{CAS: 0, Config: *config}
+	h.server.bucketDbConfigs[h.db.Bucket.GetName()] = &dbc
+	h.server.dbConfigs[dbName] = &dbc
```

Comment:
I don't think it should be the responsibility of external callers to know they need to update both internal maps - seems easy to have a case where one is missed.  Maybe an internal function on server that also manages the locking?